contrib/executor/jmeterd/pkg/slaves/slave_utils.go

```diff
@@ -0,0 +1,212 @@
+)
+
+const (
+	defaultSlavesCount = 1
+	serverPort         = 1099
+	localPort          = 60001
+)
+
+func getSlaveRunnerEnv(envParams envs.Params, runnerExecution testkube.Execution) []v1.EnvVar {
+
+	gitEnvs := []v1.EnvVar{}
+	if runnerExecution.Content.Type_ == "git" && runnerExecution.Content.Repository.UsernameSecret != nil && runnerExecution.Content.Repository.TokenSecret != nil {
+		gitEnvs = append(gitEnvs, v1.EnvVar{
+
+			Name: "RUNNER_GITUSERNAME",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: runnerExecution.Content.Repository.UsernameSecret.Name,
+					},
+					Key: runnerExecution.Content.Repository.UsernameSecret.Key,
+				},
+			},
+		}, v1.EnvVar{
+			Name: "RUNNER_GITTOKEN",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: runnerExecution.Content.Repository.TokenSecret.Name,
+					},
+					Key: runnerExecution.Content.Repository.TokenSecret.Key,
+				},
+			},
+		},
+		)
+	}
+
+	return append([]v1.EnvVar{
+		{
+			Name:  "RUNNER_ENDPOINT",
+			Value: envParams.Endpoint,
+		}, {
+			Name:  "RUNNER_ACCESSKEYID",
+			Value: envParams.AccessKeyID,
+		}, {
+			Name:  "RUNNER_SECRETACCESSKEY",
+			Value: envParams.SecretAccessKey,
+		}, {
+			Name:  "RUNNER_TOKEN",
+			Value: envParams.Token,
+		}, {
+			Name:  "RUNNER_BUCKET",
+			Value: envParams.Bucket,
+		}, {
+			Name:  "RUNNER_SSL",
+			Value: fmt.Sprintf("%v", envParams.Ssl),
+		}, {
+			Name:  "RUNNER_SCRAPPERENABLED",
+			Value: fmt.Sprintf("%v", envParams.ScrapperEnabled),
+		}, {
+			Name:  "RUNNER_DATADIR",
+			Value: envParams.DataDir,
+		}, {
+			Name:  "RUNNER_CLOUD_MODE",
+			Value: fmt.Sprintf("%v", envParams.CloudMode),
+		}, {
+			Name:  "RUNNER_CLOUD_API_KEY",
+			Value: envParams.CloudAPIKey,
+		}, {
+			Name:  "RUNNER_CLOUD_API_TLS_INSECURE",
+			Value: fmt.Sprintf("%v", envParams.CloudAPITLSInsecure),
+		}, {
+			Name:  "RUNNER_CLOUD_API_URL",
+			Value: envParams.CloudAPIURL,
+		},
+	}, gitEnvs...)
+}
+
+func getSlaveConfigurationEnv(slaveEnv map[string]testkube.Variable) []v1.EnvVar {
+	envVars := []v1.EnvVar{}
+	for envKey, t := range slaveEnv {
+		envVars = append(envVars, v1.EnvVar{Name: envKey, Value: t.Value})
+	}
+	return envVars
+}
+
+func getSlavePodConfiguration(testName string, runnerExecution testkube.Execution, envVariables map[string]testkube.Variable, envParams envs.Params) (*v1.Pod, error) {
+	runnerExecutionStr, err := json.Marshal(runnerExecution)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-jmeter-slave", testName),
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyAlways,
+			InitContainers: []v1.Container{
+				{
+					Name:            fmt.Sprintf("%s-init-container", testName),
+					Image:           "kubeshop/testkube-init-executor:1.13.12",
```

Comment:
should not be hardcoded

Reply:
Had discussion around it.
Agreed to hard-code it to latest (1.14.3) version
We will change this when we add slavesConfigs in executor CRD and add env variables for init and slave images to propagate it in executor pod.